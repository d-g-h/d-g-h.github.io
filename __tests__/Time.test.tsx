/**
 * @jest-environment jsdom
 */
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Time from "@/components/Time";

const START = "2020-02";
const END = "2024-02";

describe("Time", () => {
  it("renders Time", () => {
    render(<Time start={START} end={END}></Time>);

    const start = screen.getByText("February 2020");
    const end = screen.getByText("February 2024");

    expect(start).toBeInTheDocument();
    expect(end).toBeInTheDocument();
  });
  it("renders Time without end date", () => {
    render(<Time start={START}></Time>);

    const present = screen.getByText("- present");

    expect(present).toBeInTheDocument();
  });
  it("renders html correctly", () => {
    const { container } = render(<Time start={START} end={END}></Time>);
    expect(container).toMatchSnapshot();
  });
});
