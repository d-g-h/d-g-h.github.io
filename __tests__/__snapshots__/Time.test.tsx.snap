// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Time renders html correctly 1`] = `
<div>
  <time
    datetime="2020-02"
  >
    February 2020
  </time>
  <span>
     - 
  </span>
  <time
    datetime="2024-02"
  >
    February 2024
  </time>
</div>
`;
